### ISSUE TITLE
document root issues

### ISSUE THREAD

**Issue by Symply13** [GrahamDumpleton/mod_wsgi#526]
Hi there!  

Running, Windows10, WAMP3.2.0 64, apache 2.4.41, python 3.8.1 64.  

My flask app is supposed to accept uploaded files to "document root"/static/files/..  

When I run the flask development server and test uploads, files are uploaded and displayed as supposed to in the correct folder....  

However when I run the application with Apache and mod_wsgi, the files are being uploaded to the defined APACHE_DIR(per httpd.conf)/static/files/

not even in the default document root "C:\wamp64\www\"

web.wsgi:
> #!/usr/bin/python
> import sys
> sys.path.insert(0, 'C:/vvv/vvv/vvv/vvv/FlaskWeb/')
> from FlaskWeb import app as application

httpd-vhosts.conf:
> <VirtualHost *:80>
>         ServerName  localhost
>         WSGIScriptAlias / "C:/vvv/vvv/vvv/vvv/FlaskWeb/web.wsgi"
>         DocumentRoot "C:/vvv/vvv/vvv/vvv/FlaskWeb/"
>         <Directory "C:/vvv/vvv/vvv/vvv/FlaskWeb/">
>                 Require all granted
>         </Directory>
>         ErrorLog "C:/vvv/vvv/vvv/vvv/FlaskWeb/logs/error.log"
>         CustomLog "C:/vvv/vvv/vvv/vvv/FlaskWeb/logs/access.log" common
> </VirtualHost>

httpd.conf was not edited other than adding the mod_wsgi required lines to it.  

Only thing i can see pointing to the directory the files end up in httpd.conf: 
> Define APACHE_DIR ${INSTALL_DIR}/bin/apache/apache${VERSION_APACHE}

Thank you for helping me figure this out!

**Comment by GrahamDumpleton** [comment]
Where is your Flask configuration/code which says where files are to be uploaded? Ensure you aren't using a relative path when specifying the directory.

**Comment by Symply13** [comment]
Graham!  Thanks for responding so fast!  You're awesome!  

Flask code said the files are to be uploaded r'FlaskWeb/static/files'.  because I was under the assumption that the document root for the entire app would be "/" 
after switching it to /vvv/vvv/vvv/vvv/FlaskWeb/static/files the files landed in the correct folder.  

to be able to NOT write it with the file system os path.  I would have to start a venv and then the path would be relative to the venv directory right?

How do you activate the virtual environment for windows environment?  windows users activate the venv by running activate.bat from within the venv/scripts folder... The docs mention an activate_this.py (which windows users do not get).  How to configure web.wsgi to activate the env first?

**Comment by GrahamDumpleton** [comment]
See:

* https://modwsgi.readthedocs.io/en/develop/user-guides/application-issues.html#application-working-directory

for suggestions of how to calculate an absolute path at runtime relative to your code base.

Details on how to activate a virtual environment when using mod_wsgi is explained in:

* https://modwsgi.readthedocs.io/en/develop/user-guides/virtual-environments.html

The recommend method of activating a virtual environment on Windows is using ``WSGIPythonHome``.

**Comment by Symply13** [comment]
Hey Graham, 

I had to step away from the venv issue and come back.  Working on it again today... 

Where do I put WSGIPythonHome?  
I ask because when running "mod_wsgi-express module-config", it gives back: 

> (venv) C:\Python36-64\venv\Scripts>mod_wsgi-express module-config
> LoadFile "c:/python36-64/python36.dll"
> LoadModule wsgi_module "c:/python36-64/lib/site-packages/mod_wsgi/server/mod_wsgi.cp36-win_amd64.pyd"
> **WSGIPythonHome** "c:/python36-64"

So I put the above into httpd.conf and the app runs, however it's not getting the environ vars set within the venv.  

I tried updating httpd.conf to: "c:/python36-64/venv/FlaskWeb" and "c:/python36-64/venv/Scripts" 

Apache starts but when trying to access the web page I get an endless loop which doesn't load..  Switched it back to how config-module recommends and add WSGIPythonHome = 'C:/python36-64/venv/FlaskWeb' to web.wsgi file,  

The app runs, but still no environ vars..  Documentation mentions VirtualHosts..  So I tried adding it to httpd-vhosts.conf (outside of the virtualhost definition) and Apache does not start. I put it within the virtual host def and still does not start.

I'm on windows and cannot use WSGIDaemonProcess myapp python-home=/usr/local/venvs/myapp

What am I doing wrong?

Thanks again!
